### DISCUSSION TITLE
What is the difference between deleted version and an old version that was updated from new version ?

### DISCUSSION THREAD

**Discussion by AmadeuAntunes** [OrchardCMS/OrchardCore#8755] [Q&A]
I would like to know what is difference of a deleted content and and not published content.

|Name | Content Item Id | Content Item Version Id|  Latest | Published | Desc |
| -----------| ----------- | ----------- | ----------- |----------- | ----------- |
| A | 4h9xg9cn1467krn96a31wybtxe| 4r63pacvtaqfw2xagwhfdf745m| true | true| This version was created |
| B | 4h9xg9cn1467krn96a31wybtxe| 4ra4z7qf8zmjt308pd9sfd9a4d| true | true| Just updated the version **A** and this one was generated |
| A | 4h9xg9cn1467krn96a31wybtxe| 4r63pacvtaqfw2xagwhfdf745m| **false**| **false**| Just updated the version **A** and this one changed |
| B | 4h9xg9cn1467krn96a31wybtxe| 4ra4z7qf8zmjt308pd9sfd9a4d| **false**| **false**| Deleted version **B**

The name is a friendly way of distinguishing **content item versions** 
This data comes from database json table **[Document]** and column **[Content]**

Version B is deleted but version A never was deleted but updated, how do I know the difference between B deleted and  A that never was requested a delete?

**Comment by jtkech** [comment]
A ContentItem with a given ContentItemId may have multiple records with different ContentItemVersionId

A version that is neither latest, nor published is an inactive version

A ContentItem with a given ContentItemId is considered as soft deleted if there are no more active version

Otherwise, normally it has only one latest version, and can't have more than one published version.

E.g. if a ContentItem has a version that is latest an published, when you edit it and save it as a draft, and if Versionable, we create a new draft version that becomes the latest, the previous one is no more the latest but still published.

E.g. if a ContentItem has a version that is latest an published, when you edit it and publish it, and if Versionable, we still create a new version that becomes the latest and published, the previous one is no more the latest and no more published (so inactive).

Your above example is in an inconsistent state as for the content item id you have 2 versions that are latest and punlished.

**Reply by AmadeuAntunes** [reply]
if no version of content is published it means the content was deleted?

and if one content is not the last and isn't published but there are some content published it means is inactive?

so If I want to show all versions if the content isn't deleted will I need to check if is there some content published ?

How can I know if the content is just draft?

**Reply by jtkech** [reply]
> if no version of content is published it means the content was deleted?

Yes soft deleted

> and if one content is not the last and isn't published but there are some content published it means is inactive?

Yes, i is an inactive version

> so If I want to show all versions if the content isn't deleted will I need to check if is there some content published ?

You may want to show all active versions, those that are latest or published or both.

If no active versions, this means that the content item was soft deleted

If you only filter on the content item id you will get all versions, active or not, as the future audit trail module will manage.

> How can I know if the content is just draft?

A draft version is an active version that is latest but not published

A published version is an active version that is published but that can be latest or not
